tests/PHPStan/Analyser/data/bug-1239.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace Bug1239;
+
+use function PHPStan\Testing\assertType;
+
+class Foo
+{
+	public function __construct()
+	{
+		if (get_class($this) === Bar::class) {
```

Comment:
just working on one of the different calltypes of bug 1239.
will work on the other types in separate PRs